torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp

```diff
@@ -362,6 +362,34 @@ void checkCalculateQParamsResult(const IValue& qparams) {
   }
 }
 
+class ProcessWeightObserverSubgraph {
+ public:
+  // Given a list of nodes in src, produce a Graph with these nodes.
+  void buildObserverSubgraph(
+      const std::vector<Node*>& src,
+      std::shared_ptr<Graph> dest);
+
+  // Recursively find the nodes that produce the value and add to subgraph.
+  void findSubgraph(Value* self, Value* v, std::vector<Node*>& weight_subgraph);
+
+  // Given a list of nodes, build a graph corresponding to these nodes.
+  // User should make sure to run this graph with expected input.
+  std::unique_ptr<GraphFunction> buildGraphFromNodes(
+      const std::vector<Node*>& nodes,
+      const std::string& name);
+
+ private:
+  // Clone node in the destination Graph g.
+  void cloneNodeInGraph(Node* node, std::shared_ptr<Graph>& g);
+
+  // Create a new value corresponding to v in the Graph g.
+  Value* updateInputValueInGraph(Value* v, std::shared_ptr<Graph>& g);
+
+  // Map from values in original graph to corresponding values in the extracted
+  // subgraph.
+  std::unordered_map<Value*, Value*> remap_values_in_observer_subgraph_;
```

Comment:
is it possible to remove this as a member variable?

Reply:
Checked couple of other uses of graph clone functions in the codebase, they seem to have this as member variable.

Reply:
are you sure? I only saw local variable

Reply:
e.g. https://codebrowser.bddppq.com/pytorch/pytorch/torch/csrc/jit/codegen/cuda/graph_fuser.cpp.html#104,
https://codebrowser.bddppq.com/pytorch/pytorch/torch/csrc/jit/passes/loop_unrolling.cpp.html#73,
https://codebrowser.bddppq.com/pytorch/pytorch/torch/csrc/jit/passes/utils/subgraph_utils.cpp.html#158

Reply:
Here is the example I was referring to. https://codebrowser.bddppq.com/pytorch/pytorch/torch/csrc/jit/passes/bailout_graph.cpp.html#215. I think either way works. In this case since we are using different functions for each step we have a member variable. If we have one large function doing everything then it can be local.

Reply:
yeah, one large function sounds good

Reply:
I like the modularity of this class with the individual functions. If you are talking about only combining updateInputValueInGraph and cloneNodeInGraph, then I am fine either way.